TST: test_float_sum failure

This test fails for me locally and indeed you should almost never assert exact equality between floating point values, so I'll open a PR with a simple fix shortly:

```
____________________________________________________________________________________________________________________ TestKokkosFunctionsTranslator.test_float_sum _____________________________________________________________________________________________________________________

self = <tests.test_kokkosfunctions_translator.TestKokkosFunctionsTranslator testMethod=test_float_sum>

    def test_float_sum(self):
        expected_result: float = self.threads * (self.f_1 + self.f_2)
        result: float = pk.parallel_reduce(self.range_policy, self.functor.add_floats)
    
>       self.assertEqual(expected_result, result)
E       AssertionError: 340.0 != 340.0000000000003

tests/test_kokkosfunctions_translator.py:168: AssertionError

```

--- a/tests/test_kokkosfunctions_translator.py
+++ b/tests/test_kokkosfunctions_translator.py
@@ -166,7 +166,7 @@ class TestKokkosFunctionsTranslator(unittest.TestCase):
         expected_result: float = self.threads * (self.f_1 + self.f_2)
         result: float = pk.parallel_reduce(self.range_policy, self.functor.add_floats)
 
-        assert_allclose(expected_result, result)
+        assert_allclose(result, expected_result)
 
     def test_nested_sum(self):
         expected_result: int = self.threads * (self.i_1 + self.i_2)

--- a/tests/test_kokkosfunctions_translator.py
+++ b/tests/test_kokkosfunctions_translator.py
@@ -1,6 +1,7 @@
 import unittest
 
 import pykokkos as pk
+from numpy.testing import assert_allclose
 
 
 # Tests translation of KOKKOS_FUNCTIONS to C++
@@ -165,7 +166,7 @@ class TestKokkosFunctionsTranslator(unittest.TestCase):
         expected_result: float = self.threads * (self.f_1 + self.f_2)
         result: float = pk.parallel_reduce(self.range_policy, self.functor.add_floats)
 
-        self.assertEqual(expected_result, result)
+        assert_allclose(expected_result, result)
 
     def test_nested_sum(self):
         expected_result: int = self.threads * (self.i_1 + self.i_2)

